lua/packer/compile.lua

```diff
@@ -158,9 +158,19 @@ local function dump_loaders(loaders)
 end
 
 local function make_try_loadstring(item, chunk, name)
-  local bytecode = string.dump(item, true)
-  local executable_string = 'try_loadstring(' .. vim.inspect(bytecode) .. ', "' .. chunk .. '", "'
-                              .. name .. '")'
+  local executable_string, bytecode
+  if type(item) == 'string' then
+    bytecode = item
+    local function_body = fmt(
+[[(function()
+  %s
```

Comment:
I don't use the string format at all but does this work if my condition was `vim.env.VAR ~= nil` this would just be an expression, it would need to be returned right.

Reply:
Sorry I couldn't understand what you ment there .

> if my condition was vim.env.VAR ~= nil this would just be an expression

Are you talking about condition option here ? If so this is for config & setup option .

Basically if you had
```lua
use{ 'author/plugin',
config = 'vim.env.VAR ~= nil'
}
```

Config portion of It used to get compiled to:
```lua
vim.env.VAR ~= nil
```
Now it compiles to:
```lua
try_loadstring([[(function()
  vim.env.VAR ~= nil
end)()]], "plugin", "config")
```

And we do error check at `try_loadstring`

Either way I can't understand what `vim.env.VAR ~= nil` suppose to do here :P

Reply:
@shadmansaleh what I was alluding to is how is that expression evaluated by packer if it isn't returned? You need to return inside the function otherwise it just returns nil?

Reply:
Packer doesn't care about return value of config/setup . They are just ran for side effects as per my understanding

Reply:
Ah sorry @shadmansaleh I'm being dumb, need more sleep. Yeah was thinking of the `cond` fields and not paying enough attention.